addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -212,18 +243,113 @@ def _peppol_get_message_status(self):
                     {'message_uuids': list(message_uuids.keys())},
                 )
 
-    def _cron_peppol_get_participant_status(self):
-        edi_users = self.search([('company_id.account_peppol_proxy_state', '=', 'pending')])
-        edi_users._peppol_get_participant_status()
-
     def _peppol_get_participant_status(self):
         for edi_user in self:
             try:
                 proxy_user = edi_user._make_request(
-                    f"{edi_user._get_server_url()}/api/peppol/1/participant_status")
+                    f"{edi_user._get_server_url()}/api/peppol/2/participant_status")
             except AccountEdiProxyError as e:
                 _logger.error('Error while updating Peppol participant status: %s', e)
                 continue
 
-            if proxy_user['peppol_state'] in {'active', 'rejected', 'canceled'}:
+            if proxy_user['peppol_state'] in ('receiver', 'rejected'):
                 edi_user.company_id.account_peppol_proxy_state = proxy_user['peppol_state']
+
+    # -------------------------------------------------------------------------
+    # BUSINESS ACTIONS
+    # -------------------------------------------------------------------------
+
+    def _peppol_register_sender_as_receiver(self):
+        self.ensure_one()
+
+        company = self.company_id
+        edi_identification = self._get_proxy_identification(company, 'peppol')
+
+        if company.account_peppol_proxy_state != 'sender':
+            # a participant can only try registering as a receiver if they are currently a sender
+            peppol_state_translated = dict(company._fields['account_peppol_proxy_state'].selection)[company.account_peppol_proxy_state]
+            raise UserError(
+                _('Cannot register a user with a %s application', peppol_state_translated))
+
+        if (
+            not company.account_peppol_migration_key
+            and company.partner_id._check_peppol_participant_exists(edi_identification)
+        ):
+            raise UserError(
+                _("A participant with these details has already been registered on the network. "
```

Comment:
You are getting this error even in Demo mode. Is it normal?

Reply:
I didn't receive this error in demo mode :thinking: